### DISCUSSION TITLE
Equivalent of clearing all inputs and states

### DISCUSSION THREAD

**Discussion by ndarilek** [Leafwing-Studios/leafwing-input-manager#219] [Q&A]
I have [this](https://github.com/lightsoutgames/bevy_input_actionmap/blob/main/src/lib.rs#L331) in bevy_input_actionmap for basically nuking input state from orbit. Right now for instance I have Return bound to using powerups. Starting a game works fine because no player exists with the required components, but resuming a game always tries to launch/deploy a powerup because the bundle exists and the keybinding matches. Currently I'm using `release_all()`, but I'm guessing this doesn't quite wipe away everything.

Also, I'm hitting the state transition issue where resuming my game from the start menu via Escape immediately pauses the game because it resumes, then leafwing-input-manager detects that Escape was just pressed and re-pauses. I can only resume the game by pressing Enter on the Resume button, which again launches a powerup.

Is there a way for me to aggressively clear input state on certain transitions?

Thanks again.

**Comment by zicklag** [comment]
You can disable an input action kind entirely by using the [`ToggleActions`](https://docs.rs/leafwing-input-manager/latest/leafwing_input_manager/plugin/struct.ToggleActions.html) resource.

Essentially just initialize the resource with `app.init_resource::<ToggleActions<MyActionKind>>()` and then in your systems you can borrow it and disable/enable it like this:

```rust
fn my_system(toggle_my_action_kind: Res<ToggleActions<MyActionKind>>) {
  toggle_my_action_kind.enabled = false;
}
```

I'm not sure if that will clear the input state or not, though, so you might have to test it out and see how it interacts.

**Reply by ndarilek** [reply]
Thanks, this certainly helps in the simple case. But I have multiple input mappings exported by multiple plugins, and multiple places where I need to basically wipe input state and start over. That's a lot of use of "multiple" to make a point--it's going to get complex real quick if I have to remember to add each new map I create at each point where I need this functionality. Better, I think, to just wipe the slate clean and let the various states repopulate if I can.
